src/ctimer_linux.cpp

```diff
@@ -92,6 +92,10 @@ Error CTimer::start(Arguments& args) {
         return Error("Could not set pthread hook");
     }
 
+    if (args._target_cpu != -1) {
+        return Error("target-cpu != -1 is not supported by CTimer CPU sampling");
+    }
```

Comment:
Since the option is specific to PerfEvents engine, instead of checking it in every other engine, you may check it globally like [here](https://github.com/async-profiler/async-profiler/blob/5ea64a15c362f62f39fab4a7b8ca095b18da3121/src/profiler.cpp#L1169-L1170). This will also reduce the amount of changes.

Reply:
a21d908797d6025ce88177ac337ae9e872308d05